### ISSUE TITLE
CORS issues running static site hosted on S3

### ISSUE THREAD

**Issue by ahkai86** [MeRahulAhire/S3-Ultimate-Guide#2]
Hi Rahul, its me again! Sry to bother as i tried a few days fixing the CORS issues on hosting static website on S3 bucket.
Kept encountering this issues when i tried Multipart Upload (above 5GB)

"has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource."

i am still fixing this issues.

**Comment by MeRahulAhire** [comment]
I do have a CORS section in video [check here](https://youtu.be/6VHc41idHZs?t=1382). 

In nutshell, you'll need to enter this in your s3 bucket CORS configuration on aws console 
```js
[
{
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "PUT",
            "POST",
            "DELETE"
        ],
        "AllowedOrigins": [
            "*" 
        ],
        "ExposeHeaders": []
    }
]
```
[For more info, check this](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ManageCorsUsing.html)

**Comment by ahkai86** [comment]
Hi Rahul, Good Day! I have already set the CORS on both S3 bucket (static file host bucket + storing uploaded files bucket), but i still encounter error. Using chrome:

Access to XMLHttpRequest at 'https://xxxxxxxxxxx.execute-api.ap-southeast-1.amazonaws.com/dev/getUploadId/getUploadId' from origin 'http://xxxxxxxxxxxxxxx-southeast-1.amazonaws.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

**Comment by ahkai86** [comment]
For your frontendMultiPartUpload.js API call, is it 'https://xxx.execute-api.ap-southeast-1.amazonaws.com/dev/getUploadId` or otherwise?

**Comment by ahkai86** [comment]
Hi Rahul, i solved it by changing the API in frontendMultiPartUpload.js to "/dev" instead, and also input CORs on both S3 buckets

**Comment by ahkai86** [comment]
error still occurs after setting the CORS, any other specific settings do i need to input apart from the S3 bucket??

[
{
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "PUT",
            "POST",
            "DELETE"
        ],
        "AllowedOrigins": [
            "*" 
        ],
        "ExposeHeaders": []
    }
]

**Comment by ahkai86** [comment]
error still occurs after setting the CORS, any other specific settings do i need to input apart from the S3 bucket??

[
{
"AllowedHeaders": [
""
],
"AllowedMethods": [
"PUT",
"POST",
"DELETE"
],
"AllowedOrigins": [
""
],
"ExposeHeaders": []
}
]

**Comment by MeRahulAhire** [comment]
there is no `"*"` in your headers . copy paste below
```js
[
{
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "PUT",
            "POST",
            "DELETE"
        ],
        "AllowedOrigins": [
            "*" 
        ],
        "ExposeHeaders": []
    }
]
```

**Comment by ahkai86** [comment]
Hi Rahul, managed to get it solved for CORS! Thanks so much! 

But now for multipart upload (POST) have another issue:
POST https://xxxxxxxx-api.ap-southeast-1.amazonaws.com/dev/completeUpload 500
(anonymous) @ isAxiosError.js:10
e.exports @ isAxiosError.js:10
e.exports @ isAxiosError.js:10
Promise.then (async)
r.request @ isAxiosError.js:10
r.<computed> @ isAxiosError.js:10
(anonymous) @ isAxiosError.js:10
(anonymous) @ frontendMultiPartUpload.js:56
async function (async)
(anonymous) @ frontendMultiPartUpload.js:10

**Comment by ahkai86** [comment]
Hi Rahul, think i solved it with the new CORS policy:

[
    {
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "PUT",
            "POST",
            "DELETE"
        ],
        "AllowedOrigins": [
            "*"
        ],
        "ExposeHeaders": [
            "ETag"
        ]
    }
]

**Comment by MeRahulAhire** [comment]
I've mentioned all these in my video. Feel free to watch it.
